CONTRIBUTING.md

```diff
@@ -12,19 +12,282 @@
+
+Quickstarts need to be in a separate directories from snippets because they need
+to be in different packages.
+
+The output can be verified during testing using a buffer.
+
+[inspect_test.go](https://github.com/GoogleCloudPlatform/golang-samples/blob/readme/dlp/snippets/inspect/inspect_test.go)
+```go
+func TestInspectString(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	buf := new(bytes.Buffer)
+	err := inspectString(buf, tc.ProjectID, "I'm Gary and my email is gary@example.com")
+	if err != nil {
+		t.Errorf("TestInspectFile: %v", err)
+	}
+
+	got := buf.String()
+	if want := "Info type: EMAIL_ADDRESS"; !strings.Contains(got, want) {
+		t.Errorf("got %q, want %q", got, want)
+	}
+}
+```
+
+## Prefer inline proto declarations
+
+Where possible, prefer a single declaration when initializing a proto value.
+
+Request values should usually be named `req` and be declared on their own so the
+API call (which uses `req`) is easier to understand.
+
+```diff
+- myRequest := &pb.Request{}
+- myRequest.Parent = projectID
++ req := &pb.Request{
++ 	Parent: projectID,
++ }
+```
+
+## Initialize clients and services in every sample
+
+Don't initialize one client for the entire set of samples and pass it as an
+argument. Each sample should initialize its own client/service.
+
+```diff
+- func hello(client foo.Client, w io.Writer) { ... }
++ func hello(w io.Writer) {
++	client, err := foo.NewClient(...)
++	// ...
++ }
+```
+
+## Return errors
+
+If the sample can run into errors, return the errors with additional context.
+Don't call `log.Fatal` or friends.
+
+`log.Fatal` is difficult to test because it will stop the entire test suite.
+
+Use `fmt.Errorf` to add information when returning errors. Usually, the name of
+the `package.Function` or just `Function` that returned the error is enough. It
+may also help to include any arguments that were passed to the function.
+
+Prefer inline error declaration when they aren't needed outside the `if`
+statement.
+
+```diff
+client, err := foo.NewClient()
+// delete deletes the resource identified by resourceID.
+func delete(w io.Writer, resourceID string) error {
+	// resource := fmt.Sprintf("/projects/my-project/resources/%s", resourceID)
+	client, err := foo.NewClient()
+-	if err != nil {
+-		log.Fatal(err)
+-	}
+-	err := client.Delete(resourceID)
+-	if err != nil {
+-		log.Fatal(err)
+-	}
++	if err != nil {
++		return fmt.Errorf("foo.NewClient: %v", err)
++ 	}
++	if err := client.Delete(resourceID); err != nil {
++		return fmt.Errorf("Delete: %v", err)
++	}
+	return nil
+}
+```
+
+
+## Imports
+
+Imports should be added and sorted by `goimports`. There should be two groups,
+separated by a newline:
+* Standard library
+* Everything else
+
+```go
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"golang.org/x/exp/rand"
+)
+```
+
+## Comment functions and packages
```

Comment:
any guideline on naming for packages and their directory names?

Reply:
Added ("Sample package name, file name, and directory").